lib/browser/api/web-contents.ts

```diff
@@ -641,6 +641,7 @@ WebContents.prototype._init = function () {
         url,
         frameName,
         features: rawFeatures,
+        parsedFeatures: parseCommaSeparatedKeyValue(rawFeatures),
```

Comment:
why this and not `parseFeatures`? `parseFeatures` seems to have special behavior for `left` and `top`. Should that be moved into `parseCommaSeparatedKeyValue`? It's also a bit surprising that certain keys are converted to numbers, while others are left as strings. It's also unclear from the docs how this handles eg. duplicate keys.

... so basically, i'm not sure this is the right thing to do? Can you expand more on what the use case is here?

Reply:
we are missing the parsed options when migrating from the deprecated `new-window` event to `setWindowOpenHandler`.

Reply:
before you could do this for example (a bit artificial one)
```js
  mainWindow.webContents.on('new-window', (event, url, frameName, disposition, options) => {
    options.width = Math.min(640, options.width);
    options.height = Math.min(480, options.height);
  });
```

Reply:
which needs to be replaced by something like this
```js
  mainWindow.webContents.setWindowOpenHandler(details => {
    return {
      action: 'allow',
      overrideBrowserWindowOptions: {
        width: Math.min(640, details.parsedFeatures.width),
        height: Math.min(640, details.parsedFeatures.height),
      }
    };
  });
```

Reply:
`new-window` gives us the calculated `BrowserWindowConstructorOptions` that we can modify. However with `setWindowOpenHandler` we can just override values, but we don't know what they are until we get the `did-create-window` event.

Reply:
I am also not sure whether to use `parseFeatures` or `parseCommaSeparatedKeyValue` here.

Reply:
if the goal is to be able to intercept what Electron "would" use for those features, perhaps we can expose an object that's only the "meaningful" features? i.e. width and height would be there, but any custom stuff would be absent. we could expose the parsed version of anything electron would use, and everything else we'd drop on the floor in the `parsedFeatures` structure.

if we do something more generic, we should document how we handle duplicate values, syntax confusion (e.g. what do feature strings `,,,` or `=` or `===` or `,,=` produce?), whitespace handling (e.g. is `foo=bar` the same as `foo= bar` or `foo=bar ` etc.), and any other parsing quirks.